src/cpp/session/modules/SessionConnections.R

```diff
@@ -517,4 +517,34 @@ options(connectionObserver = list(
    .rs.success()
 })
 
+.rs.addJsonRpcHandler("connection_test", function(code) {
+   error <- ""
 
+   oldConnections <- lapply(ls(.rs.activeConnections), function(name) {
+      get(name, envir = .rs.activeConnections)
+   })
+
+   .envir <- parent.frame(2)
+   tryCatch({
+      eval(parse(text = code), envir = .envir)
+   }, error = function(e) {
+      error <<- e$message
+   })
+
+   if (nchar(error) == 0) {
+      newConnections <- lapply(ls(.rs.activeConnections), function(name) {
+         get(name, envir = .rs.activeConnections)
+      })
+
+      oldConnectionIds <- lapply(oldConnections, function(x) paste(x$type, x$host, sep = "|"))
```

Comment:
It will be hard to undo all the side effects of having the IDE pick up and store the connection; for instance, it will still wind up in the connection history, which we don't want if it's not a valid connection. 

Instead I think we should have the IDE completely ignore connections which are created just to test connectivity. This could happen here:

https://github.com/rstudio/rstudio/blob/0aaa25abe98e8a48f6135f984c2471dcf889dfe0/src/cpp/session/modules/SessionConnections.R#L75

We could use the environment or assigned variable name as a flag to indicate a test connection, or just check to see if we're inside this function.